service/history/queue_factory_base.go

```diff
@@ -219,6 +220,55 @@ func (f *QueueFactoryBase) Stop() {
 	}
 }
 
+func (f *QueueFactoryBase) NewExecutableFactory(
+	shardContext shard.Context,
```

Comment:
plz don't use shardContext as a property bag for passing dependencies. 

I actually wonder if having this method on queueFactoryBase will save us any code. Each queue still need to call this method and pass those params in.

Reply:
> plz don't use shardContext as a property bag for passing dependencies.

Definitely agree here--this was pure laziness. Fixed.

> I actually wonder if having this method on queueFactoryBase will save us any code. Each queue still need to call this method and pass those params in.

It saves us just `f.HostScheduler`, `f.HostPriorityAssigner`, and `if executableWrapper == nil {
	return factory
}`, so I think it's worth it.

Reply:
sounds good. but I will make a per shard delayed rate limited task scheduler, so that will be a param as well :)